src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -59,24 +59,54 @@ public class HubEndPoint<THub, TClient> : IInvocationBinder where THub : Hub<TCl
 
         public async Task OnConnectedAsync(ConnectionContext connection)
         {
-            await ProcessNegotiate(connection);
+            var output = Channel.CreateUnbounded<byte[]>();
+            var connectionContext = new HubConnectionContext(output, connection);
+
+            await ProcessNegotiate(connectionContext);
```

Comment:
> The sever has settting per endpoint that determines what to send (binary or text) and there's no way to do this in a per connection level. Maybe that's what you mean? Assume we have a way to set that in some connection metadata or feature so that it can be read by the websocket transport.
Yes, not being able to set it per connection is the first blocker. And yes, I wanted to set it on Metadata (only it felt quite gross).

> For SSE why would the client bother to send a request to the server if we ended up with an invalid combination, say "protobuf and SSE" or msgpack and SSE?
- you request Transport.All and wanted to use ProtoBuf. Then you run on Win7 and you get the best available transport which is SSE. You end up with SSE and binary. 
- another problem is that if it is not supported the server has not enough information to reject invalid combinations if this is not supported